File: 1/2020.09.22/49.Group Anagrams/java/solution.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();
        for (String str : strs) {
            char[] ca = new char[26];
            for (char c : str.toCharArray()) {
                ca[c-'a']++;
            }
            String sortedStr = String.valueOf(ca);
            if (!map.containsKey(sortedStr)) {
                List<String> list = new ArrayList<>();
                map.put(sortedStr, list);
            }
            map.get(sortedStr).add(str);
        }
        return new ArrayList<>(map.values());
    }

}

File: 1/2020.09.22/49.Group Anagrams/python/solution.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        d = {}
        for str in strs:
            key = ''.join(sorted(str))
            if key not in d:
                d[key] = []
            d[key].append(str)
        return list(d.values())
